### PR TITLE
STEP-3: Add functionality to Delete tasks

### PR DESCRIPTION
**Pull request by sta-wtag**
[STEP-3](https://trello.com/c/xM9zRilm/2-step-3-add-functionality-to-delete-tasks
)
Delete a task when clicked on the delete icon of a task.

### **Changes**
1. Delete task from store when clicked on the delete button.
2. UI should update itself when a tasks is deleted.

### CODE REVIEW

File: components/TaskCard.vue [hunk 1]
```diff
@@ -11,13 +11,14 @@
       <div>
         <EditButton />
       </div>
-      <div>
+      <div @click="deleteTask()">
```
Comment (hhu-wtag) [inline components/TaskCard.vue]: Please remove the function invocation as you are immediately calling the function.
```suggestion
      <div @click="deleteTask">
```

File: store/mutations.js [hunk 2]
```diff
@@ -2,6 +2,17 @@ export default {
...
+  deleteTask(state, val) {
+    const list = state.states.taskList;
+
+    for (let i = 0; i < list.length; i++) {
+      if (list[i].id === val.id) {
+        const index = list.indexOf(list[i]);
+
+        list.splice(index, 1);
+      }
+    }
```
Comment (hhu-wtag) [inline store/mutations.js]: While imperative way of doing things is fine, it's suggested that we try to code in a declarative way. 
```suggestion
   state.states.taskList = state.states.taskList.filter(task => task.id !== val.id)
```

File: components/TaskCard.vue [hunk 3]
```diff
@@ -30,6 +31,18 @@ export default {
       default: null,
     },
   },
+  data: () => ({
+    task: '',
```
Comment (hhu-wtag) [inline components/TaskCard.vue]:
```suggestion
    task: null,
```

File: components/TaskCard.vue [hunk 4]
```diff
@@ -30,6 +31,18 @@ export default {
...
+    loading: false,
+  }),
+  mounted() {
+    this.task = _.clone(this.cardData);
```
Comment (hhu-wtag) [inline components/TaskCard.vue:39]: Why do we need to clone this here?

**Review by hhu-wtag — CHANGES_REQUESTED** [review]
Requested some changes.

File: components/TaskCard.vue (hunk 4 above)
Comment (sta-wtag) [reply components/TaskCard.vue:39]: It throughs an error saying i am editing the vuex state outside the store while i try to edit the task description

File: store/mutations.js (hunk 2 above)
Comment (sta-wtag) [reply store/mutations.js:14]: I am also removing the item from the list after filtering it out

File: components/TaskCard.vue (hunk 1 above)
Comment (sta-wtag) [reply components/TaskCard.vue]: Resolved

File: components/TaskCard.vue (hunk 3 above)
Comment (sta-wtag) [reply components/TaskCard.vue]: Resolved

File: store/mutations.js (hunk 2 above)
Comment (sta-wtag) [reply store/mutations.js]: Resolved

File: components/TaskCard.vue (hunk 4 above)
Comment (hhu-wtag) [reply components/TaskCard.vue:39]: Are you by any chance trying to mutate store state inside a component?

**Review by bia-wtag — APPROVED** [review]
Tested locally. LGTM.

**Pull request merged**